tests/main/snapd-reexec/task.yaml

```diff
@@ -0,0 +1,78 @@
+summary: Test that snapd rexecs itself into core
+
+systems: [-ubuntu-core-16-64, -ubuntu-core-16-arm-64, -ubuntu-core-16-arm-32]
+
+environment:
+    SNAPD_CONF: /etc/systemd/system/snapd.service.d/local.conf 
+
+execute: |
+    if [ "${SNAP_REEXEC:-}" = "0" ]; then
+        echo "skipping test when SNAP_REEXEC is disabled"
+        exit 0
+    fi
+
+    echo "Ensure we re-exec by default"
```

Comment:
We currently write into syslog that we restart and into what path. Do you think that is sufficient?